fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -936,9 +995,23 @@ protected SqlQueryData processInclusionCriteria(QueryParameter queryParm) throws
                 // (pX.PARAMETER_NAME_ID = x AND
                 this.populateNameIdSubSegment(whereClauseSegment, currentParm.getCode(), PARAMETER_TABLE_ALIAS);
                 whereClauseSegment.append(AND);
-                // Build this piece: pX.str_value = search-attribute-value
-                whereClauseSegment.append(PARAMETER_TABLE_ALIAS + DOT).append(STR_VALUE).append(operator)
+                
+                // split the resource type name out (since issue #1366)
+                String resourceTypeName = null;
+                String[] parts = currentParmValue.split("/");
+                if (parts.length == 2) {
+                    resourceTypeName = parts[0];
+                    currentParmValue = parts[1];
+                }
+                
+                // Build this piece: pX.token_value = search-attribute-value [ AND pX.code_system_id = <n> ]
+                whereClauseSegment.append(PARAMETER_TABLE_ALIAS + DOT).append(TOKEN_VALUE).append(operator)
```

Comment:
shouldn't this be split into two appends? `.append(PARAMETER_TABLE_ALIAS).append(DOT)`

Reply:
yes